### PR TITLE
fix: make internalSetMaxConsumersPerSubscription async

### LINKED ISSUE #15856: make internalSetMaxConsumersPerSubscription async
**Describe the bug**

make internalSetMaxConsumersPerSubscription async

**To Reproduce**
1. create and delete topic frequently
**Expected behavior**

you will see there will a waiting forever in internalDeleteTopicForcefully

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by leizhiyuan**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #15856

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by leizhiyuan** [comment]
> @mattisonchao We need to be connected to #15043 ?

I didn't notice that before, It seems that someone is working on this issue.  maybe I can wait for that

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 1]
```diff
@@ -2201,28 +2201,27 @@ protected Integer internalGetMaxConsumersPerSubscription() {
...
+                        throw new RestException(Status.PRECONDITION_FAILED,
+                                "maxConsumersPerSubscription must be 0 or more");
+                    }
+                    return updatePoliciesAsync(namespaceName, policies -> {
+                        policies.max_consumers_per_subscription = maxConsumersPerSubscription;
+                        return policies;
+                    });
+
+                }).exceptionally(ex -> {
+                    Throwable realCause = FutureUtil.unwrapCompletionException(ex);
```
Comment (shibd) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:2220]: Exceptions can be handled in the rest layer

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java [hunk 2]
```diff
@@ -1251,11 +1251,21 @@ public Integer getMaxConsumersPerSubscription(@PathParam("property") String prop
...
+            @Suspended final AsyncResponse asyncResponse,
             @PathParam("property") String property,
             @PathParam("cluster") String cluster,
             @PathParam("namespace") String namespace, int maxConsumersPerSubscription) {
         validateNamespaceName(property, cluster, namespace);
-        internalSetMaxConsumersPerSubscription(maxConsumersPerSubscription);
+        internalSetMaxConsumersPerSubscription(maxConsumersPerSubscription).thenAccept(
+                        __ -> asyncResponse.resume(Response.noContent().build()))
+                .exceptionally(ex -> {
+                    if (!isRedirectException(ex)) {
```
Comment (shibd) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:1262]: The original logic does not have this judgment condition.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java (hunk 1 above)
Comment (leizhiyuan) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java]: done

**Comment by leizhiyuan** [comment]
Since it takes a long time and the code differences are too large, I will close this PR.

**Pull request closed without merging**